src/components/page/WithGATracker/index.tsx

```diff
@@ -10,6 +10,19 @@ import store from '../../../store';
 type Props = RouteComponentProps<RouteParams>;
 let username = (getUser(store.getState()) || {}).username || '';
 
+/**
+ * helper function to get the Google Analytics tracking code
+ * @param {FlexObject|undefined} options tracking options for the page view
+ * @returns {string}
+ */
+export const getGaCode = (options: FlexObject | undefined) => {
+  const gaCode =
+    (options && !!!options.GA_CODE && options.GA_CODE) ||
+    ((!options || !!options.GA_CODE) && GA_CODE) ||
+    '';
+  return gaCode;
+};
+
```

Comment:
The intent was to default to the global `env. GA_CODE` value if the `options.GA_CODE` was not provided, so that we don't need to pass the `GA_CODE` to every single route in `App.tsx`. If you think it's better we can just pass it in every time to simplify this and make `options` a required param.

Reply:
I think we should consider getting GA_CODE from options only.  It would make the function [easier to reason about and more flexible](https://stackify.com/solid-design-open-closed-principle/).

Reply:
It is so, thanks for the read!